apps/projection/src/app/component/teacher-card/teacher-card.component.ts

```diff
@@ -1,36 +1,57 @@
-import { Component, OnInit } from '@angular/core';
-import { FakeHttpService } from '../../data-access/fake-http.service';
+import { AsyncPipe } from '@angular/common';
+import {
+  ChangeDetectionStrategy,
+  Component,
+  OnInit,
+} from '@angular/core';
+import { map, Observable } from 'rxjs';
+import {
+  FakeHttpService,
+  randTeacher,
+} from '../../data-access/fake-http.service';
 import { TeacherStore } from '../../data-access/teacher.store';
-import { CardType } from '../../model/card.model';
-import { Teacher } from '../../model/teacher.model';
+import { Item } from '../../model/item.model';
 import { CardComponent } from '../../ui/card/card.component';
 
 @Component({
   selector: 'app-teacher-card',
   template: `<app-card
-    [list]="teachers"
-    [type]="cardType"
+    [items]="teacherItems$ | async"
     customClass="bg-light-red"
-  ></app-card>`,
-  styles: [
-    `
-      ::ng-deep .bg-light-red {
-        background-color: rgba(250, 0, 0, 0.1);
-      }
-    `,
-  ],
+    [listItemTemplate]="listItemTemplate"
+    (addNewItem)="addNewTeacher()"
+    (deleteItem)="deleteTeacher($event)">
+    <ng-container image>
+      <img src="assets/img/teacher.png" width="200px" />
+    </ng-container>
+    <ng-template #listItemTemplate let-item>{{ item?.name }} (T)</ng-template>
```

Comment:
nice 👍 
now you don't need to have a name property for each component, you can write whatever you want since it's projected.

And you should have you listItemcomponent here. everything will be simplier